stl/inc/atomic

```diff
@@ -1158,6 +1158,24 @@ struct _Atomic_integral<_Ty, 8> : _Atomic_storage<_Ty> { // atomic integral oper
 #endif // _M_IX86
 };
 
+#if 1 // TRANSITION, ABI
+template <size_t _TypeSize>
+_INLINE_VAR constexpr bool _Is_always_lock_free = _TypeSize <= 8 && (_TypeSize & (_TypeSize - 1)) == 0;
+#else // ^^^ don't break ABI / break ABI vvv
+#if _ATOMIC_HAS_DCAS
+template <size_t _TypeSize>
+_INLINE_VAR constexpr bool _Is_always_lock_free = _TypeSize <= 2 * sizeof(void*);
+#else // ^^^ _ATOMIC_HAS_DCAS / !_ATOMIC_HAS_DCAS vvv
+template <size_t _TypeSize>
+_INLINE_VAR constexpr bool _Is_always_lock_free = _TypeSize <= sizeof(void*);
+#endif // _ATOMIC_HAS_DCAS
+#endif // break ABI
+
+template <bool _Is_lock_free>
+_INLINE_VAR constexpr bool _Deprecate_non_lock_free_volatile = true;
+
+template <>
+_CXX20_DEPRECATE_VOLATILE _INLINE_VAR constexpr bool _Deprecate_non_lock_free_volatile<false> = true;

```

Comment:
This is explicitly specializing a variable template without `inline` in C++14 mode, which is a conformance issue that has caused problems for Clang in the past (not sure about the current state, but it was a huge headache in June 2019). @CaseyCarter, should we make this a partial specialization instead, and comment the reason why (as I very nearly forgot)?

Perhaps templating this on `_Ty` and adding a default argument `bool = _Is_always_lock_free<sizeof(_Ty)>` so we can partially specialize `<_Ty, false>` would both avoid this conformance issue and improve usability below, where we could simply say `_Deprecate_non_lock_free_volatile<_Ty>`?